### DISCUSSION TITLE
PnP Search and Results error for most internal users

### DISCUSSION THREAD

**Discussion by tsmanz** [microsoft-search/pnp-modern-search#3740] [Q&A] [answered]
**Version used**
4.11.1

**Describe the bug**
I've setup PnP modern search for an IT Knowledge Base on SharePoint Online. It's a subsite of our IT Hub site collection and is used to search for site pages which are KB articles. The search function works fine for myself and the IT team, but non-IT users have reported seeing errors rather than the search box and results.

The non-IT users have Visitor permissions for the site collection, so I tried setting my user account as a Visitor, but wasn't able to replicate the error. I then created a test user account and applied the Visitor permissions and was able to replicate the error. As a further test, I've also tried setting the test user as a Member and also as an Owner of the site collection but the error still occurs for them.

None of the users are external users, all are hybrid AD including the accounts that don't receive the error.

**Search Box Error:**
```
Something went wrong
If the problem persists, contact the site administrator and give them the information in Technical Details.
TECHNICAL DETAILS
ERROR:
Script error for: 544c1372-7e5a-49ec-8db6-812f76c375f2_4.11.0/SearchBoxWebPartStrings
http://requirejs.org/docs/errors.html#scripterror

CALL STACK:
Error: Script error for: 544c1372-7e5a-49ec-8db6-812f76c375f2_4.11.0/SearchBoxWebPartStrings
http://requirejs.org/docs/errors.html#scripterror
    at B (eval at e (https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_b579cd4edfb4a2fae3bb11eb75f7824c.js:197:10298), <anonymous>:8:252)
    at HTMLScriptElement.onScriptError (eval at e (https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_b579cd4edfb4a2fae3bb11eb75f7824c.js:197:10298), <anonymous>:30:244)
```

**Search Results Error:**
```
Something went wrong
If the problem persists, contact the site administrator and give them the information in Technical Details.
TECHNICAL DETAILS
ERROR:
Script error for: 544c1372-42df-47c3-94d6-017428cd2baf_4.11.0/SearchResultsWebPartStrings
http://requirejs.org/docs/errors.html#scripterror

CALL STACK:
Error: Script error for: 544c1372-42df-47c3-94d6-017428cd2baf_4.11.0/SearchResultsWebPartStrings
http://requirejs.org/docs/errors.html#scripterror
    at B (eval at e (https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_b579cd4edfb4a2fae3bb11eb75f7824c.js:197:10298), <anonymous>:8:252)
    at HTMLScriptElement.onScriptError (eval at e (https://res-1.cdn.office.net/files/sp-client/sp-pages-assembly_en-us_b579cd4edfb4a2fae3bb11eb75f7824c.js:197:10298), <anonymous>:30:244)
```

**To Reproduce**
Open the IT Knowledge Base site

**Expected behavior**
The site should load a PnP search box and PnP search results containing site pages from the Knowledge Base.

**Desktop:**
 - Browser Edge/Chrome

**Comment by kasperbolarsen** [comment]
Which language are you using? 
Version 4.11.0 had a bug in the danish language pack, which have been fixed in version 4.11.1 (current version).

**Comment by tsmanz** [comment]
I'm using English so not sure that update will make much difference.

**Comment by kasperbolarsen** [comment]
Please try it and revert. That error message is the same we saw related to the bug in 4.11.0

**Comment by tsmanz** [comment]
Ok will apply it tomorrow to test

**Comment by tsmanz** [comment]
I've applied the update. Unfortunately it hasn't fixed or changed the error.

**Comment by jackvandeur** [comment]
We had the same problem. We reverted to version 4.10.2.0 and cleared the cash. Customers environment is working again.

**Comment by tsmanz** [comment]
I tried reverting to 4.10.2 but unfortunately get exactly the same error.

**Comment by wobba** [comment]
Check that the the users has access to the app catalog where you have installed the .sppkg.

**Reply by tsmanz** [reply, marked as answer]
Looks like it was a permissions issue. Initially I had added the app to the tenant app catalog. I've  now created an app catalog for the site collection that uses pnp modern search, then added the latest version and set it up from there.

Thanks for pointing me in the right direction, much appreciated!
